### PR TITLE
Add revision information to service list

### LINKED ISSUE #360: have `kn service list` show latest revision name
/kind good-first-issue
/kind feature
/kind proposal

## Describe the feature:

Today `kn service list` will show:
```
NAME   URL                                                             GENERATION   AGE   CONDITIONS   READY   REASON
echo   http://echo-default.kndev.us-south.containers.appdomain.cloud   2            65s   3 OK / 3     True    
```

Notice the revision name is not there. Seeing the revision name is important for users so they can know when the latest version of their app has been deployed and ready to be hit.

See similar output from `kubectl get ksvc`:
```
NAME   URL                                                             LATESTCREATED   LATESTREADY   READY   REASON
echo   http://echo-default.kndev.us-south.containers.appdomain.cloud   echo-v1         echo-v1       True    
```

Proposal: add a `Latest Revision` or just `RevisionName` column so people can see which is the latest revision that's going to be used.

I'd drop `Generation` as that's a number that's not very useful for end-users, other than to know it's been changed, which is only true if they're tracking it. Someone like that, I suspect, would probably be naming their revisions and would be more interested in seeing this new column's value instead.

Not sure I see value in `Conditions` either since seeing `0/3` or `3/3` doesn't really give any more useful info than the overall `Ready` column - I'd drop it too.

### PR DESCRIPTION
**Pull request by daisy-ycguo**
Fixes #360

## Proposed Changes

* Add revision `LATESTCREATED` and `LATESTREADY` to the output of `kn service list`
* Remove `Generation` from the output of `kn service list`

### CODE REVIEW

File: pkg/kn/commands/service/human_readable_flags.go [hunk 1]
```diff
@@ -98,9 +100,9 @@ func printKServiceWithNaemspace(kServiceList *servingv1alpha1.ServiceList, optio
 func printKService(kService *servingv1alpha1.Service, options hprinters.PrintOptions) ([]metav1beta1.TableRow, error) {
 	name := kService.Name
 	url := kService.Status.URL
-	//lastCreatedRevision := kService.Status.LatestCreatedRevisionName
-	//lastReadyRevision := kService.Status.LatestReadyRevisionName
-	generation := kService.Status.ObservedGeneration
+	lastCreatedRevision := kService.Status.ConfigurationStatusFields.LatestCreatedRevisionName
+	lastReadyRevision := kService.Status.ConfigurationStatusFields.LatestReadyRevisionName
+	// generation := kService.Status.ObservedGeneration
```
Comment (rhuss) [inline pkg/kn/commands/service/human_readable_flags.go]: Please remove completely if not needed anymore

File: pkg/kn/commands/service/human_readable_flags.go [hunk 2]
```diff
@@ -117,9 +119,9 @@ func printKService(kService *servingv1alpha1.Service, options hprinters.PrintOpt
 	row.Cells = append(row.Cells,
 		name,
 		url,
-		//lastCreatedRevision,
-		//lastReadyRevision,
-		generation,
+		lastCreatedRevision,
+		lastReadyRevision,
+		//generation,
```
Comment (rhuss) [inline pkg/kn/commands/service/human_readable_flags.go]: Please remove

File: pkg/kn/commands/service/list_mock_test.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ func TestServiceListAllNamespaceMock(t *testing.T) {
 	assert.NilError(t, err)
 
 	outputLines := strings.Split(output, "\n")
-	assert.Assert(t, util.ContainsAll(outputLines[0], "NAMESPACE", "NAME", "URL", "GENERATION", "AGE", "CONDITIONS", "READY", "REASON"))
+	assert.Assert(t, util.ContainsAll(outputLines[0], "NAMESPACE", "NAME", "URL", "LATESTCREATED", "LATESTREADY", "AGE", "CONDITIONS", "READY", "REASON"))
```
Comment (maximilien) [inline pkg/kn/commands/service/list_mock_test.go]: Would these two words read better was `LATEST CREATED` and `LATEST READY` assuming that's not what's exactly coming from APIs?

**Comment by daisy-ycguo** [comment]
@rhuss @maximilien Thank you for the reviewing.
I refer to the output of `kubectl get ksvc` to design the output of `kn service list`. `LATESTCREATED` and `LATESTREADY ` are displayed in the output of `kubectl get ksvc`.

Considering your comments, I'm going to keep only lastReadyRevision as `LATEST REVISION` in the output of `kn service list`. Please let me know if it's OK.

```
$ kubectl get ksvc
NAME          URL                                                                          LATESTCREATED       LATESTREADY         READY     REASON
fib-knative   http://fib-knative-default.knative-guoyc.au-syd.containers.appdomain.cloud   fib-knative-x7zj9   fib-knative-x7zj9   True
```

File: pkg/kn/commands/service/human_readable_flags.go (hunk 2 above)
Comment (daisy-ycguo) [reply pkg/kn/commands/service/human_readable_flags.go]: done

File: pkg/kn/commands/service/list_mock_test.go (hunk 3 above)
Comment (daisy-ycguo) [reply pkg/kn/commands/service/list_mock_test.go]: Change to `LATEST REVISION`

**Comment by daisy-ycguo** [comment]
I think it's ready for a second round review. @maximilien @rhuss Thanks.

File: pkg/kn/commands/service/human_readable_flags.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ func ServiceListHandlers(h hprinters.PrintHandler) {
 		{Name: "Namespace", Type: "string", Description: "Namespace of the Knative service", Priority: 0},
 		{Name: "Name", Type: "string", Description: "Name of the Knative service.", Priority: 1},
 		{Name: "Url", Type: "string", Description: "URL of the Knative service.", Priority: 1},
-		{Name: "Generation", Type: "integer", Description: "Sequence number of 'Generation' of the service that was last processed by the controller.", Priority: 1},
+		{Name: "Latest Revision", Type: "string", Description: "Name of the latest revision.", Priority: 1},
```
Comment (navidshaikh) [inline pkg/kn/commands/service/human_readable_flags.go]: Should we have space in column names? Can we change the column name to `Latest` as suggested in review?
Also per changes, the description should read, "Name of the latest ready revision."

Comment (daisy-ycguo) [reply pkg/kn/commands/service/human_readable_flags.go]: Changed to `Latest`. The comment is changed to `Name of the latest ready revision.`.

**Comment by daisy-ycguo** [comment]
thank you @navidshaikh and @rhuss . It's ready to review again.

**Pull request merged**